keywords.go

```diff
@@ -145,6 +160,41 @@ var (
 		"link",
 		"tar_time",
 	}
+
+	// BsdKeywords is the set of keywords that is only in the upstream FreeBSD mtree
+	BsdKeywords = []string{
+		"cksum",
+		"device",
+		"flags",
+		"ignore",
+		"gid",
+		"gname",
+		"link",
+		"md5",
+		"md5digest",
+		"mode",
+		"nlink",
+		"nochange",
+		"optional",
+		"ripemd160digest",
+		"rmd160",
+		"rmd160digest",
+		"sha1",
```

Comment:
ah, i was going by https://www.freebsd.org/cgi/man.cgi?mtree(8) which shows
`sha1` as a supported keyword

On Tue, Jul 26, 2016 at 4:12 PM, Stephen Chung <notifications@github.com>
wrote:

> In keywords.go
> <https://github.com/vbatts/go-mtree/pull/45#discussion_r72327402>:
>
> > +		"device",
> > +		"flags",
> > +		"ignore",
> > +		"gid",
> > +		"gname",
> > +		"link",
> > +		"md5",
> > +		"md5digest",
> > +		"mode",
> > +		"nlink",
> > +		"nochange",
> > +		"optional",
> > +		"ripemd160digest",
> > +		"rmd160",
> > +		"rmd160digest",
> > +		"sha1",
>
> [schung@localhost Desktop] $ gomtree -c -p smalldir -k sha1 > smalldir.mtree
> [schung@localhost Desktop] $ cat smalldir.mtree
> #          user: schung
> #       machine: localhost.localdomain
> #          tree: smalldir
> #          date: Tue Jul 26 16:08:50 2016
>
> # .
> /set type=file uid=1000 gid=1000
> . type=dir
>     file1 sha1=da39a3ee5e6b4b0d3255bfef95601890afd80709
>     file2 sha1=da39a3ee5e6b4b0d3255bfef95601890afd80709
>     file3 sha1=da39a3ee5e6b4b0d3255bfef95601890afd80709
> ..
> [schung@localhost Desktop] $ mtree -f smalldir.mtree -p smalldir
> mtree: line 9: unknown keyword sha1
> [schung@localhost Desktop] $ mtree -c -p smalldir -k sha1digest
> #      user: (null)
> #   machine: localhost.localdomain
> #      tree: /home/schung/Desktop/smalldir
> #      date: Tue Jul 26 16:09:20 2016
>
> # .
> /set type=file
> .               type=dir
>     file1       sha1digest=da39a3ee5e6b4b0d3255bfef95601890afd80709
>     file2       sha1digest=da39a3ee5e6b4b0d3255bfef95601890afd80709
>     file3       sha1digest=da39a3ee5e6b4b0d3255bfef95601890afd80709
> ..
>
> gomtree currently supports keywords sha1 and sha1digest, so to resolve
> conflicts when validating with an upstream mtree spec, obtaining the
> keywordfuncs for both sha1 and sha1digest in gomtree should return the
> same function.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vbatts/go-mtree/pull/45/files/919d71c1058b75672cac768ea24db1731dffc930#r72327402>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEF6ckqbNgLee8X81TJw4wBVt5d2btNks5qZmpBgaJpZM4JVc_r>
> .
>